docs/modules/ROOT/pages/leader-election.adoc

```diff
@@ -25,3 +25,76 @@ To specify the name of the configmap used for leader election use the following
 ----
 spring.cloud.kubernetes.leader.config-map-name=leader
 ----
+
+'''
+
+There is another way you can configure leader election and it comes with native support in the fabric8 library. In the long run, this will be the default way to configure leader election, while the previous one will be dropped. You can treat this one much like the JDKs "preview" features.
+
+To be able to use it, you first need to disable the old implementation:
+
+[source]
+----
+spring.cloud.kubernetes.leader.enabled=false
+----
+
+and enable the new one:
+
+[source]
+----
+spring.cloud.kubernetes.leader.election.enabled=true
+----
+
+Unlike the old implementation, this one will use either the `Lease` _or_ `ConfigMap` as the lock, depending on your cluster version. The name of that `Lease` or `ConfigMap` can be defined using the property (default value is `spring-k8s-leader-election-lock`):
+
+[source]
+----
+spring.cloud.kubernetes.leader.election.lockName=other-name
+----
+
+The namespace (`default` being set if no explicit one exists) can be set also:
```

Comment:
Shouldnt the namespace be the namespace the pod is running in?

Reply:
But wouldn't it make sense for the default location for the lock to be created be the namespace the pods are running not 'default'?